internal/workflow/preprocessing.go

```diff
@@ -77,6 +106,20 @@ func (w *PreprocessingWorkflow) Execute(
 		&activities.ValidateFileFormatsParams{ContentPath: identifySIP.SIP.ContentPath},
 	).Get(ctx, &validateFileFormats)
 
+	// Add PREMIS event noting validate file formats result.
+	e = temporalsdk_workflow.ExecuteActivity(
+		withLocalActOpts(ctx),
+		activities.AddPREMISEventName,
+		&activities.AddPREMISEventParams{
+			Path:  localPath,
+			Agent: premis.PREMISAgentDefault(),
+			Type:  "validateFileFormats",
+			Error: validateFileFormatsErr},
+	).Get(ctx, &addPREMISEvent)
+	if e != nil {
+		return nil, e
+	}
+
```

Comment:
I think we may want to do some of these updates to the PREMIS files from inside the existing activities. That will give you access to the actual file formats and how the validation went for each file while it's happening.

Reply:
I've reworked the validate file formats activity to add individual per-file events.